add pull secret flag

mjudeikis opened the pull request.

m1kola commented on docs/using-az-aro.md:
```diff
@@ -140,6 +140,7 @@ cluster:
      --vnet dev-vnet \
      --master-subnet "$CLUSTER-master" \
      --worker-subnet "$CLUSTER-worker"
+     --pull-secret $PULL_SECRET
```
I think the step of setting `$PULL_SECRET` is missing from this doc.

mjudeikis replied:
Yes, its part of our `./env`. Will update

jim-minter commented on hack/e2e/run-rp-and-e2e.sh:
```diff
@@ -114,7 +114,8 @@ run_e2e() {
       --client-id $CLUSTER_SPN_ID \
-      --client-secret $CLUSTER_SPN_SECRET
+      --client-secret $CLUSTER_SPN_SECRET \
+      --pull-secret $PULL_SECRET
```
should have double quotes around this and other arguments

jim-minter commented:
@asalkeld this was the PR I meant

mjudeikis commented:
> This needs more thought.
> 
>     * Let's clear out any end user documentation from this repo and immediately redirect end users to the official docs from README.md.  Let's not worry about end user documentation any more in this repo.  Possibly a separate PR.

Ok, not touching it for now

> 
>     * For developers, PULL_SECRET must be set up to run the RP, and it should contain only the static ACR token.
>       (If that PULL_SECRET also has the RH tokens, then --pull-secret is irrelevant.  Alternatively if it doesn't have the RH tokens, then --pull-secret behaves like it does for end users.)
>       To be close as possible to user behaviour, we should probably then have a second pull secret env var (USER_PULL_SECRET?) which is a standard RH pull secret.  We tell developers to do `--pull-secret "$USER_PULL_SECRET"` if they want to.  I don't mind much whether the e2e does that or not.

Split both tokens into `USER_PULL_SECRET` and `PULL_SECRET`.  And updated secrets.

m1kola commented:
@mjudeikis @jim-minter  User documentation for preview is already here: https://docs.microsoft.com/en-us/azure/openshift/howto-using-azure-redhat-openshift

jim-minter commented:
> Ok, not touching it for now

Open an issue to track please?

jim-minter commented:
Repushed with some text from https://github.com/MicrosoftDocs/azure-docs/pull/50890
